controller/src/server/functions.ts

```diff
@@ -268,10 +285,33 @@ export async function queueRun(run_id:string):Promise<string> {
   return 'ok';
 }
 
+function connectHasuraEndpoint():void {
+  const hasura = process.env.HASURA ?? 'http://127.0.0.1:8080/v1/graphql';
+  client = new GraphQLClient(hasura, {
+    headers: {
+      'x-hasura-admin-secret': 'hasuraadminsecret',
```

Comment:
This should probably be but in a `.env` file that is not in the repository, and use `process.env.HASURA_ADMIN_SECRET` or similar instead.